### ISSUE TITLE
Add Mastodon Handle to profile

### ISSUE THREAD

**Issue by eszter007** [bookwyrm-social/bookwyrm#2566]
I would like to add my Mastodon handle with the rel=me tag to my Bookwyrm profile:
<a rel="me" href="https://example-instance@account-name">Mastodon</a>
If I add then my Bookwyrm profile link into my Mastodon bio, it will appear as verified (on Mastodon). I want my followers to know that the Bookwyrm profile is truely mine.

**Comment by jaschaurbach** [comment]
You can already do this in the profile text - it takes HTML tags. Maybe the documentation should be updated regarding this. 

<img width="1179" alt="Bildschirm­foto 2023-01-09 um 17 50 18" src="https://user-images.githubusercontent.com/2017105/211362810-e52b8c76-913a-478e-bde5-45b4a2d3c4df.png">

**Comment by mouse-reeve** [comment]
I wonder if bookwyrm should adopt the profile lists that mastodon has now?

**Comment by jaschaurbach** [comment]
But why? With the free field and the rendering of HTML it is far more powerful than a list.

**Comment by mouse-reeve** [comment]
That's true, and the profile lists may be a bit off topic to this actual request. But not every user knows or should need to know HTML and markdown, and it would be nice to have more support for that. But as you said, we do currently support rel=me links! And updating the docs would be a good idea.

**Comment by legoktm** [comment]
Could bare links in the profile field automatically be marked up with `rel="me"`? 

Currently if I type `I am https://wikis.world/@legoktm` it turns into `<p>I am <a href="https://wikis.world/@legoktm">wikis.world/@legoktm</a></p>`. It's already parsing it to turn it into an `<a>` tag, can it add `rel="me"` at that step?

> You can already do this in the profile text - it takes HTML tags. Maybe the documentation should be updated regarding this.

There could be help text below or above the box along the lines of "Your profile may use some HTML markup".

**Comment by asmaloney** [comment]
> Could bare links in the profile field automatically be marked up with rel="me"?

I don't think that's a good idea. `rel="me"` is specific to the thing being linked to, and you don't want it on most links.

Back to the original post, I would favour a semi-structured approach that would be easy for any non-technical people to use. Maybe a limited list of "known sites" they can link to in a dropdown along with a free-form option. We can treat the "known sites" specially - like including logos/colour in buttons for them. Maybe it needs a way to indicate "this is me" which adds `rel="me"` to the link? 

So If I have a Mastodon account, I could select "Mastodon" and then paste in either the URL or my mastodon id. This would show up on my profile as a Mastodon button (pill/buttonlet/tag - I don't know the right web term for this?) which points at my Mastodon profile. These could show up in a row below the profile text or below the avatar + data section.

Similar workflow for other "known site" links. What goes in the list? Not sure exactly because you can't enumerate them all... which is why you would allow free-form metadata-style links.

**Comment by legoktm** [comment]
> > Could bare links in the profile field automatically be marked up with rel="me"?
> 
> I don't think that's a good idea. `rel="me"` is specific to the thing being linked to, and you don't want it on most links.

Because rel="me" is bi-directional, I don't think that matters as much in practice. On Mastodon, whatever URL you put in your profile is marked up with rel="me", regardless of whether it's actually me. But that has no effect unless the other site is also linking back with rel="me".

And for links you really don't want marked up, you could write an explicit `<a href="...">...</a>` without the rel="me"?

**Comment by hughrun** [comment]
I agree with @mouse-reeve it's probably not great practice to assume every link is a `rel="me"`.

I like @asmaloney's suggestion of an interface specifically for personal links, however a drop-down list is inevitably going to either by unreasonably long and/or the cause of endless requests to add sites to it. I'm uncomfortable with privileging any particular sites or platforms over others. I think simply offering unlimited free-text options would work nicely (similar to how we deal with multiple author entry).

**Comment by asmaloney** [comment]
> I think simply offering unlimited free-text options would work nicely (similar to how we deal with multiple author entry).

I'd be fine with that. The only reason I suggested having some "known" sites in there is to make it easy/obvious for tech-challenged people. "(sees list) Oh! I have a mastodon account. I can put that in here." And then have it show up nicely in the UI.

**Comment by jaschaurbach** [comment]
Maybe we just need a better doc or some explanation on the site how to insert a link which can verify with a mastodon account? Just thinking...

**Comment by eszter007** [comment]
> Maybe we just need a better doc or some explanation on the site how to insert a link which can verify with a mastodon account? Just thinking...

I would not expect a user to read the documentation here from a UX perspective. I am with @asmaloney here to have a solution which is easy to understand for non-tech-savvy people, like having a list of known sites they can link to we show them nicely in the UI with a section socials e.g. and a bunch of icons (e.g. Mastodon, Pixelfed, Instagram, ...)

**Comment by jaschaurbach** [comment]
> I would not expect a user to read the documentation here from a UX perspective.

Sorry, I wasn‘t clear. With „on the site“ I meant on the page where the form is. When a user clicks „edit“ they get the form and above or beyond should be an explanation.

By the way: we can expect users to read a userguide which at the moment us too few user-documentation and too much admin doc (maybe we shouldn’t split hat). but if  there were a user guide and best linked from the corresponding site, it is reasonable to expect users to read it.

**Comment by asmaloney** [comment]
[Here's](https://github.blog/changelog/2023-02-02-add-more-social-links-to-your-user-profile/)`s what GitHub has done for this. Similar to Mastodon.

I like that it's free-form, but recognizes some sites so they show up in a user-friendly way in the profile:

![Screen Shot 2023-02-02 at 10 36 11](https://user-images.githubusercontent.com/391371/216369800-9313bfe8-77cf-4fc5-98c5-0d22763712c2.png)
